src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1024,6 +930,18 @@ MasterIrreducibleExpressionWalker(Node *expression, WalkerState *state)
 }
 
 
+static bool
+MasterIrreducibleExpressionFunctionChecker(Oid func_id, void *context)
```

Comment:
> Doesn't declaring this directly as bool *volatileContext work?

No it does not.  See definitions in nodeFuncs.h

```c
typedef bool (*check_function_callback) (Oid func_id, void *context);

...

extern bool check_functions_in_node(Node *node, check_function_callback checker,
				    void *context);

extern bool expression_tree_walker(Node *node, bool (*walker) (),
				   void *context);
```

```expression_tree_walker``` allows functions with arbitrary ```coercible``` type but ```check_functions_in_node``` does not due to difference in functions' declaration.